stream/src/main/scala/org/apache/pekko/stream/javadsl/SubSource.scala

```diff
@@ -2246,8 +2247,9 @@ class SubSource[Out, Mat](
    *
    * '''Cancels when''' downstream cancels
    */
-  def zipWithIndex: javadsl.SubSource[pekko.japi.Pair[Out @uncheckedVariance, Long], Mat] =
-    new SubSource(delegate.zipWithIndex.map { case (elem, index) => pekko.japi.Pair(elem, index) })
```

Comment:
too much boxing for Java users, that's unfair.

Reply:
<img width="945" alt="image" src="https://github.com/user-attachments/assets/2edab18b-547b-4e69-8d67-9af675506c46" />
Fixed now.